### PR TITLE
feat: added functional for the add endpoint

### PR DESCRIPTION
**Pull request by NicholasBoren**
Implemented add endpoint and two helper methods.
All tests checked by Elliot.

### CODE REVIEW

File: api/server.js [hunk 1]
```diff
@@ -154,9 +154,33 @@ app.get("/bookshelves/get/:username/:bookshelf", async function(req, res) {
...
+            if (userId == 0) {
+                res.status(CLIENT_ERROR_CODE_401).send("Invalid username");
+			} else if (!isValidBookshelf) {
+				res.status(CLIENT_ERROR_CODE_400).send("Invaild bookshelf name");
+			} else if (!isValidISBN) {
+				res.status(CLIENT_ERROR_CODE_400).send("Book does not exist"); 
+			} else if (await helper.checkIfBookExistsInBookshelf(bookshelf, userId, isbn)) {
+                res.status(CLIENT_ERROR_CODE_400).send("Book already exists in " + bookshelf);
+            } else {
+                let tableAltered = await helper.insertBook(bookshelf, userId, isbn);
```
Comment (ASimpleFox) [inline api/server.js:177]: you could just do await helper.insertBook, no need for let tableAltered = await...

**Review by ASimpleFox — CHANGES_REQUESTED** [review]
Fix the one lil issue in server.js and we should be good to go! (you could also check the deepscan thing and fix any issues it finds yourself) ;D

File: api/server.js (hunk 1 above)
Comment (NicholasBoren) [reply api/server.js]: Fixed the issue!

**Review by ASimpleFox — APPROVED** [review]
Looks good! I assume this is all tested!

**Comment by huskycodingproject** [comment]
All endpoints should be completed and tested, all that's left is some final debugging, connecting back-end with front-end, and resolving any issues that the front-end might find.

**Pull request merged**